dev/snippets/lib/snippets.dart

```diff
@@ -202,6 +202,11 @@ class SnippetGenerator {
   /// The [type] is the type of snippet to create: either a
   /// [SnippetType.application] or a [SnippetType.sample].
   ///
+  /// [showDartPad] indicates whether DartPad should be shown where possible.
+  /// Currently, this value only has an effect if [type] is
```

Comment:
Instead of it only having an effect if the type is `application`, you should have it exit in `main` with an error if `--dartpad` is specified, but the type isn't application.  Failing silently is bad here, since it'll just zoom past it in the doc creation, any nobody will ever notice until they look at the broken docs.

This function should also have the same assert as above:
```dart
  assert(!showDartPad || type == SnippetType.application, 'Only application snippets work with dartpad.');
```

Reply:
Definitely a good idea. Made both of these changes.